noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/shared_mutable.nr

```diff
@@ -225,6 +225,34 @@ impl<T, INITIAL_DELAY> SharedMutable<T, INITIAL_DELAY, &mut PrivateContext> wher
     }
 }
 
+impl<T, INITIAL_DELAY> SharedMutable<T, INITIAL_DELAY, UnconstrainedContext> where T: ToField + FromField + Eq {
+    unconstrained pub fn get_current_value_in_unconstrained(self) -> T {
+        let block_number = self.context.block_number() as u32;
+        self.read_value_change().get_current_at(block_number)
+    }
```

Comment:
To avoid too much proliferation and maintenance burden I'd consider supporting only this getter and removing the rest - I imagine in unconstrained execution we'll only care about the current value and not about delays, scheduled values, etc. This would greatly cut down the number of tests we need to include.

Of course at some point we'll want a more full-fledged API, but I think it may be a bit early for that (we'll likely not use those methods for a while).